pkg/statistics/handle/globalstats/topn.go

```diff
@@ -138,7 +262,173 @@ func MergeGlobalStatsTopNByConcurrency(
 //     just delete some values when we merge the global-level topN.
```

Comment:
Could you please add the algorithm logic in the comments ~
(Especially the pruning algorithm)

Reply:
added